From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HeathLoganCampbell <idk@gmail.com>
Date: Sun, 6 Nov 2022 13:13:17 -0300
Subject: [PATCH] Netty IP_TOS 0x18


diff --git a/src/main/java/net/minecraft/server/ServerConnection.java b/src/main/java/net/minecraft/server/ServerConnection.java
index 3401addb448a9c731aec768edbe1008e9df80d17..a0ef0a9247bf1aa8af349445db8b8d023ee2a167 100644
--- a/src/main/java/net/minecraft/server/ServerConnection.java
+++ b/src/main/java/net/minecraft/server/ServerConnection.java
@@ -110,6 +110,7 @@ public class ServerConnection {
                 protected void initChannel(Channel channel) throws Exception {
                     try {
                         channel.config().setOption(ChannelOption.TCP_NODELAY, Boolean.valueOf(true));
+                        channel.config().setOption(ChannelOption.IP_TOS, 0x18); // PandaSpigot - Netty IP_TOS 0x18
                     } catch (ChannelException channelexception) {
                         ;
                     }
