From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HeathLoganCampbell <idk@gmail.com>
Date: Sun, 6 Nov 2022 12:57:43 -0300
Subject: [PATCH] Only fire InventoryCloseEvent if inventory is open


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index afd730b429fac34467176ef038d58d6e03487a0d..217e6a24abdbab8ba4e863e720a95e81b1bda690 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1458,9 +1458,12 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
         if (this.player.dead) return; // CraftBukkit
         PlayerConnectionUtils.ensureMainThread(packetplayinclosewindow, this, this.player.u());
 
-        CraftEventFactory.handleInventoryCloseEvent(this.player); // CraftBukkit
-
-        this.player.p();
+        // PandaSpigot start - Only fire InventoryCloseEvent if inventory is open
+        if (packetplayinclosewindow.getId() == player.activeContainer.windowId) {
+            CraftEventFactory.handleInventoryCloseEvent(this.player); // CraftBukkit
+            this.player.p();
+        }
+        // PandaSpigot end
     }
 
     public void a(PacketPlayInWindowClick packetplayinwindowclick) {
