From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HeathLoganCampbell <idk@gmail.com>
Date: Sun, 6 Nov 2022 13:35:15 -0300
Subject: [PATCH] Option to disable enchantment table ticking


diff --git a/src/main/java/com/hpfxd/pandaspigot/config/PandaSpigotConfig.java b/src/main/java/com/hpfxd/pandaspigot/config/PandaSpigotConfig.java
index f419b852012b8994ef870368ed63fb2d9f38d73b..d0d6d7a720cfcdad6e310403287da136ce8fb11f 100644
--- a/src/main/java/com/hpfxd/pandaspigot/config/PandaSpigotConfig.java
+++ b/src/main/java/com/hpfxd/pandaspigot/config/PandaSpigotConfig.java
@@ -130,4 +130,6 @@ public class PandaSpigotConfig {
 
     @Comment("That default number, 2097152L, puts an INSANE ~2MB limit on NBT. We changed the limit to ~0.05 MB.")
     public long nbtReadLimiter = 50000L;
+
+    public boolean shouldTickEnchantmentTables = true;
 }
diff --git a/src/main/java/net/minecraft/server/TileEntityEnchantTable.java b/src/main/java/net/minecraft/server/TileEntityEnchantTable.java
index e05416659102f126cfbd964e0c64988900bc2a05..50fdbecb339140fb79c41948ad254475ce83625c 100644
--- a/src/main/java/net/minecraft/server/TileEntityEnchantTable.java
+++ b/src/main/java/net/minecraft/server/TileEntityEnchantTable.java
@@ -36,6 +36,11 @@ public class TileEntityEnchantTable extends TileEntity implements IUpdatePlayerL
     }
 
     public void c() {
+        // PandaSpigot start - Option to disable enchantment table ticking
+        if (!com.hpfxd.pandaspigot.config.PandaSpigotConfig.get().shouldTickEnchantmentTables) {
+            return;
+        }
+        // PandaSpigot end
         this.k = this.j;
         this.m = this.l;
         EntityHuman entityhuman = this.world.findNearbyPlayer((double) ((float) this.position.getX() + 0.5F), (double) ((float) this.position.getY() + 0.5F), (double) ((float) this.position.getZ() + 0.5F), 3.0D);
